### PR TITLE
Make slug translatable

### PR DESCRIPTION
**Pull request by ubrize**
Make slug translatable:
`$form->addField(new Translatable((new Slug('slug', 'name', $this->url('api', 'slug_generator')))->rules('required')));`

And:
1. Do not regenerate slug on blur every time - only when creating object for the first time.
2. Do not append random id if it is not necessary.

### CODE REVIEW

File: src/Admin/Form/Fields/Renderer/TranslatableFieldRenderer.php [hunk 1]
```diff
@@ -108,7 +109,13 @@ protected function getLocalizationSwitch()
      */
     public function render()
     {
-        $block = Html::div()->addClass( 'field type-text i18n' );
+        $block = Html::div()->addClass( 'field i18n' );
+
+        if ($this->field->getFieldClass() === Slug::class) {
```
[the remark this answers is not in the document]
Comment (ubrize) [reply src/Admin/Form/Fields/Renderer/TranslatableFieldRenderer.php]: Currently field types are hardcoded in renderer, example -
 https://github.com/arbory/arbory/blob/master/src/Admin/Form/Fields/Renderer/RichtextFieldRenderer.php#L45 

Would it be sufficient to replace getFieldClass with getFieldType in Translatable?
```
public function getFieldType()
{
    return 'type-' . ucwords(get_class($this->field));
}
```

Or we should think of a more global approach?

[the remark this answers is not in the document]
Comment (ubrize) [reply src/Admin/Form/Fields/Renderer/TranslatableFieldRenderer.php]: Please see if this is sufficient - https://github.com/arbory/arbory/pull/32/commits/0a345c6dcea091d2fe8ea0f892822f30530acb2d

File: src/Admin/Form/Fields/AbstractField.php [hunk 2]
```diff
@@ -77,6 +77,15 @@ public function getNameSpacedName()
         ] );
     }
 
+    /**
+     * @return string
+     */
+    public function getFieldTypeName()
+    {
+        $class = substr(strrchr(static::class, '\\'), 1);
```
Comment (roboc) [inline src/Admin/Form/Fields/AbstractField.php]: It would be better to use `camel_case( class_basename(ArboryImage::class ) )`

Comment (ubrize) [reply src/Admin/Form/Fields/AbstractField.php]: Updated - https://github.com/arbory/arbory/pull/32/commits/9e1b5bcd29a21881a93a4d732a83ec4eef4d52af

**Pull request merged**